helix-term/src/ui/editor.rs

```diff
@@ -122,8 +124,19 @@ impl EditorView {
         // TODO: range doesn't actually restrict source, just highlight range
         let highlights: Vec<_> = match doc.syntax() {
             Some(syntax) => {
+                let scopes = theme.scopes();
+                let configs: Vec<_> = loader
+                    .language_configs_iter()
+                    .map(|x| (x, x.highlight_config(scopes).unwrap()))
+                    .collect();
                 syntax
-                    .highlight_iter(text.slice(..), Some(range), None, |_| None)
+                    .highlight_iter(text.slice(..), Some(range), None, |language| {
+                        let language = syntax::Lang::from_str(language).unwrap();
+                        let (.., highlight_config) = configs
+                            .iter()
+                            .find(|(config, ..)| config.language_id == language)?;
+                        Some(highlight_config)
+                    })
```

Comment:
The problem is that it requires a `&` but the lifetime is bound to the `Arc` object. We can use the same (unsafe) lifetime extension trick: 

https://github.com/helix-editor/helix/blob/084a8a952292521f1f7b7f4144e6823c29b85507/helix-core/src/syntax.rs#L394-L407

This compiles for me:

```rust
                let scopes = theme.scopes();
                syntax
                    .highlight_iter(text.slice(..), Some(range), None, |language| {
                        loader
                            .language_config_for_scope(language)
                            .and_then(|language_config| {
                                let config = language_config.highlight_config(scopes);
                                let config_ref = unsafe {
                                    std::mem::transmute::<_, &'static syntax::HighlightConfiguration>(
                                        config.as_ref(),
                                    )
                                };
                                Some(config_ref)
                            })
                    })
                    .collect() // TODO: we collect here to avoid holding the lock, fix later
```